### PR TITLE
Workspace svg tests

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Part of #2870

### Proposed Changes
- Converts the jsunit tests in workspace_svg_test.js to mocha 
- Refactors to be able to call workspace tests within workspace_svg tests
- Refactors out captureWarnings function for use in workspace tests as well and removes code that logs warnings to console to cleanup console messages when running tests.

### Reason for Changes
Converting to mocha explained in linked issue.
Refactor in order to reuse code.

### Test Coverage

Ran mocha tests.

### CODE REVIEW

File: tests/mocha/workspace_svg_test.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    var blocks = this.workspace.getAllBlocks(false);
+    chai.assert.equal(blocks[0].getRelativeToSurfaceXY().x, 21,
+        'Block 1 position x');
+    chai.assert.equal(blocks[0].getRelativeToSurfaceXY().y, 23,
+        'Block 1 position y');
+    chai.assert.equal(blocks[1].getRelativeToSurfaceXY().x, 21,
+        'Block 2 position x');
+    // Y separation value defined in appendDomToWorkspace as 10
+    chai.assert.equal(blocks[1].getRelativeToSurfaceXY().y,
+        23 + blocks[0].getHeightWidth().height + 10,
```
Comment (moniika) [inline tests/mocha/workspace_svg_test.js:55]: I added comment to explain where the 10 is coming from, but not whether the 10 in `appendDomToWorkspace` should be moved out into a constant.

File: tests/mocha/index.html [hunk 2]
```diff
@@ -13,14 +13,15 @@
...
+    <script src="../../blocks/math.js"></script>
   </head>
   <body>
 
     <div id="mocha"></div>
     <div id="failureCount" style="display:none" tests_failed="unset"></div>
     <script src="https://unpkg.com/chai/chai.js"></script>
     <script src="https://unpkg.com/mocha@5.2.0/mocha.js"></script>
-    <script src="https://unpkg.com/sinon/pkg/sinon.js"></script>
+    <script src="https://unpkg.com/sinon/pkg/sinon.js"></script></script>
```
Comment (rachel-fenichel) [inline tests/mocha/index.html:24]: Nit: extra closing tag here.

File: tests/mocha/workspace_svg_test.js (hunk 1 above)
Comment (rachel-fenichel) [reply tests/mocha/workspace_svg_test.js:55]: It's fine to leave it as-is.  It's not a parameter I want to make tweakable.

File: tests/mocha/workspace_test.js [hunk 3]
```diff
@@ -7,6 +7,18 @@
 suite('Workspace', function() {
   setup(function() {
     this.workspace = new Blockly.Workspace();
+  });
+
+  teardown(function() {
+    this.workspace.dispose();
+  });
+
+  // eslint-disable-next-line no-use-before-define
```
Comment (rachel-fenichel) [inline tests/mocha/workspace_test.js:16]: Why did you move the test definition out into a separate function? It's obviously intentional but I don't understand the benefit.

Comment (moniika) [reply tests/mocha/workspace_test.js:16]: I wouldn't be able to access the test in workspace_svg_test if I defined it within the workspace test suite.

File: tests/mocha/index.html (hunk 2 above)
Comment (moniika) [reply tests/mocha/index.html]: Done
